src/overlays/actors/ovl_En_Light/z_en_light.c

```diff
@@ -21,15 +27,181 @@ const ActorInit En_Light_InitVars = {
     (ActorFunc)EnLight_Update,
     (ActorFunc)EnLight_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Light/EnLight_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Light/EnLight_Destroy.s")
+typedef struct {
+    /* 0x00 */ Color_RGBA8_n primColor;
+    /* 0x04 */ Color_RGB8 envColor;
+    /* 0x07 */ u8 scale;
+} FlameParams;
+
+FlameParams D_80A9E840[] = {
+    { { 255, 200, 0, 255 }, { 255, 0, 0 }, 75 },     { { 255, 200, 0, 255 }, { 255, 0, 0 }, 75 },
+    { { 0, 170, 255, 255 }, { 0, 0, 255 }, 75 },     { { 170, 255, 0, 255 }, { 0, 150, 0 }, 75 },
+    { { 255, 200, 0, 255 }, { 255, 0, 0 }, 40 },     { { 255, 200, 0, 255 }, { 255, 0, 0 }, 75 },
+    { { 170, 255, 0, 255 }, { 0, 150, 0 }, 75 },     { { 0, 170, 255, 255 }, { 0, 0, 255 }, 75 },
+    { { 255, 0, 170, 255 }, { 200, 0, 0 }, 75 },     { { 255, 255, 170, 255 }, { 255, 50, 0 }, 75 },
+    { { 255, 255, 170, 255 }, { 255, 255, 0 }, 75 }, { { 255, 255, 170, 255 }, { 100, 255, 0 }, 75 },
+    { { 255, 170, 255, 255 }, { 255, 0, 100 }, 75 }, { { 255, 170, 255, 255 }, { 100, 0, 255 }, 75 },
+    { { 170, 255, 255, 255 }, { 0, 0, 255 }, 75 },   { { 170, 255, 255, 255 }, { 0, 150, 255 }, 75 },
+};
+
+extern Gfx D_0404D4E0[];
+extern Gfx D_05000440[];
+
+void EnLight_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+    s16 yOffset;
+
+    if (gSaveContext.gameMode == 3) {
+        // special case for the credits
+        yOffset = (this->actor.params < 0) ? 1 : 40;
+        Lights_InitType0PositionalLight(&this->posLightInfo, this->actor.posRot.pos.x,
+                                        yOffset + (s16)this->actor.posRot.pos.y, this->actor.posRot.pos.z, 255, 255,
+                                        180, -1);
+    } else {
+        yOffset = (this->actor.params < 0) ? 1 : 40;
+        Lights_InitType2PositionalLight(&this->posLightInfo, this->actor.posRot.pos.x,
+                                        yOffset + (s16)this->actor.posRot.pos.y, this->actor.posRot.pos.z, 255, 255,
+                                        180, -1);
+    }
+
+    this->lightNode = Lights_Insert(globalCtx, &globalCtx->lightCtx, &this->posLightInfo);
+    Actor_SetScale(&this->actor, D_80A9E840[this->actor.params & 0xF].scale * 0.0001f);
+    this->timer = (s32)(Math_Rand_ZeroOne() * 255.0f);
+
+    if ((this->actor.params & 0x400) != 0) {
+        this->actor.update = EnLight_UpdateSwitch;
+    }
+}
+
+void EnLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+
+    Lights_Remove(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void EnLight_UpdatePosRot(EnLight* this, GlobalContext* globalCtx) {
+    // update yaw for billboard effect
+    this->actor.shape.rot.y = func_8005A9F4(ACTIVE_CAM) + 0x8000;
```

Comment:
No review here, just a note that this gets the direction that the camera is pointing.

Reply:
its accurate to say its for the billboard effect then right? just makin sure

Reply:
Yep